### ISSUE TITLE
Provide error handling for Reboot() call

### ISSUE THREAD

**Issue by invidian** [coreos/go-systemd#387]
Right now user has no way of knowing if calling `Reboot()` succeeds or not, as no error is returned from a call. E.g. if user calling it has no permissions to do so, the call will silently fail.

This could be improved, e.g. together with accepting the context for operation in new function.

**Comment by lucab** [comment]
Good idea, a `RebootWithContext` would fit nicely. Would you like to submit a PR with that?

**Comment by invidian** [comment]
Can't promise I find time to do so, but will try :)

**Comment by invidian** [comment]
Ah, just found out there is more general issue for this: #326, not sure if it make sense to address each method individually or should we handle all of them at once.

**Comment by invidian** [comment]
Created #390 to address this.
